jcl/src/java.base/share/classes/openj9/tools/attach/diagnostics/base/DiagnosticProperties.java

```diff
@@ -0,0 +1,256 @@
+		baseProperties = props;
+	}
+
+	/**
+	 * Create an empty list of properties
+	 */
+	public DiagnosticProperties() {
+		baseProperties = new Properties();
+	}
+
+	/**
+	 * Add a property with an integer value
+	 * @param key property name
+	 * @param value property value
+	 */
+	public void put(String key, int value) {
+		baseProperties.setProperty(key, Integer.toString(value));
+	}
+
+	/**
+	 * Add a property with a String value
+	 * @param key property name
+	 * @param value property value
+	 */
+	public void put(String key, String value) {
+		baseProperties.setProperty(key, value);
+	}
+
+	/**
+	 * Add a property with an long value
+	 * @param key property name
+	 * @param value property value
+	 */
+	public void put(String key, long value) {
+		baseProperties.setProperty(key, Long.toString(value));
+	}
+
+	/**
+	 * Add a property with a boolean value
+	 * @param key property name
+	 * @param value property value
+	 */
+	public void put(String key, boolean value) {
+		baseProperties.setProperty(key, Boolean.toString(value));
+	}
+
+	private void checkExists(String key) throws IOException {
+		if (!containsField(key)) {
+			throw new IOException("key " + key + " not found"); //$NON-NLS-1$ //$NON-NLS-2$
+		}
+	}
+
+	/**
+	 * Test if the given property is present
+	 * @param key property name
+	 * @return true if the property name is found
+	 */
+	public boolean containsField(String key) {
+		return baseProperties.containsKey(key);
+	}
+
+	/**
+	 * Retrieve the value of a property and convert it to an int.
+	 * @param key name of the property
+	 * @return value of the property as an int
+	 * @throws NumberFormatException if the value is not a string representing a number.
+	 * @throws IOException  if the property is missing
+	 */
+	public int getInt(String key) throws NumberFormatException, IOException {
+		checkExists(key);
+		return Integer.parseInt(baseProperties.getProperty(key));
+	}
+
+	/**
+	 * Retrieve the value of a property and convert it to an long.
+	 * @param key name of the property
+	 * @return value of the property as a long
+	 * @throws NumberFormatException if the value is not a string representing a number.
+	 * @throws IOException  if the property is missing
+	 */
+	public long getLong(String key) throws NumberFormatException, IOException  {
+		checkExists(key);
+		return Long.parseLong(baseProperties.getProperty(key));
+	}
+
+	/**
+	 * Retrieve the value of a property and convert it to a boolean
+	 * @param key name of the property
+	 * @return value of the property as a boolean
+	 * @throws IOException  if the property is missing
+	 */
+	public boolean getBoolean(String key) throws IOException { 
+		checkExists(key);
+		return Boolean.parseBoolean(baseProperties.getProperty(key));
+	}
+
+	/**
+	 * Return a property value for the given key
+	 * @param key property name
+	 * @return property value or null if the property is not found
+	 */
+	public String getProperty(String key) {
+		return baseProperties.getProperty(key);
```

Comment:
Why does this method not include a call to `checkExists(key)`?

Reply:
There are places where the caller wants to do the checking.